File: sample-app-hd-calculator/src/main/java/com/hd/calculator/app/util/DisplayUtils.java
package com.hd.calculator.app.util;

import com.hd.calculator.app.MyApp;

public class DisplayUtils {
    //dp转px
    public static int dp2px(float dpValue) {
        return (int) (dpValue * MyApp.context.getResources().getDisplayMetrics().density + 0.5f);
    }

    //sp转px
    public static int sp2px(float spValue) {
        return (int) (spValue * MyApp.context.getResources().getDisplayMetrics().scaledDensity + 0.5f);
    }

    public static int getScreenWidth() {
        return MyApp.context.getResources().getDisplayMetrics().widthPixels;
    }
}

File: sample-app-cleaner-wood/src/main/java/com/clean/wood/utils/DisplayUtils.kt
package com.clean.wood.utils

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Point
import android.os.Build
import android.util.DisplayMetrics
import android.view.WindowManager
import com.clean.wood.WoodApp

object DisplayUtils {

    @SuppressLint("InternalInsetResource")
    fun getStatusBarHeight(context: Context): Int {
        var result = 0
        val resourceId = context.resources.getIdentifier("status_bar_height", "dimen", "android")
        if (resourceId > 0) {
            result = context.resources.getDimensionPixelSize(resourceId)
        }
        return result
    }


    @SuppressLint("InternalInsetResource")
    fun getNavigationBarHeight(context: Context): Int {
        var result = 0
        val resourceId = context.resources.getIdentifier("navigation_bar_height", "dimen", "android")
        if (resourceId > 0) {
            result = context.resources.getDimensionPixelSize(resourceId)
        }
        return result
    }

    fun dip2px(dip: Int): Int {
        val displayMetrics = WoodApp.context.resources.displayMetrics
        val density = displayMetrics.density
        return (dip * density + 0.5f).toInt()
    }

    private fun getDisplayMetrics(): DisplayMetrics {
        val resource = WoodApp.context.resources
        return resource.displayMetrics
    }

    fun getScreenWidth(): Int {
        return getDisplayMetrics().widthPixels
    }

    fun getScreenHeight(context: Context): Int {
        val windowManager = (context.getSystemService(Context.WINDOW_SERVICE) as WindowManager)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            val metrics = windowManager.currentWindowMetrics
            return metrics.bounds.height()
        } else {
            val display = windowManager.defaultDisplay
            val point = Point()
            display.getRealSize(point)
            val mScreenH = point.y
            return mScreenH
        }

    }

}